internal/s3_agent/s3_agent.go

```diff
@@ -14,7 +15,8 @@ import (
 
 // S3Agent wraps the s3.S3 structure to allow for wrapper methods
 type S3Agent struct {
-	Client *s3.S3
+	Client             *s3.S3
+	BucketAccessAction map[string][]string
```

Comment:
We can replace this field with a function that returns a variable of type `map[string][]string`. That would improve readability and would let the S3Agent still have a single responsibility of wrapping the S3 Client (and not carrying extra information).

Reply:
Done

Reply:
A nit. The newly introduced variable, `BucketAccessAction`, starts with a capital case letter.

Reply:
Done